lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -679,32 +681,32 @@ class ParseableInterfaceModuleLoaderImpl {
   // to the interface file it represents. If it's up-to-date, return its status,
   // so we can switch hash-based dependencies to mtimes after we've validated
   // the hashes.
-  Optional<llvm::vfs::Status> dependencyIsUpToDate(const FileDependency &dep) {
-    auto Status = getStatusOfDependency(fs, dep.getPath(), interfacePath,
+  bool dependencyIsUpToDate(const FileDependency &dep) {
+    auto status = getStatusOfDependency(fs, dep.getPath(), interfacePath,
                                         diags, diagnosticLoc);
-    if (!Status) return None;
+    if (!status) return false;
 
     // If the sizes differ, then we know the file has changed.
-    if (Status->getSize() != dep.getSize()) return None;
+    if (status->getSize() != dep.getSize()) return false;
 
     // Otherwise, if this dependency is verified by modification time, check
     // it vs. the modification time of the file.
     if (dep.isModificationTimeBased()) {
       uint64_t mtime =
-        Status->getLastModificationTime().time_since_epoch().count();
-      return mtime == dep.getModificationTime() ? Status : None;
+        status->getLastModificationTime().time_since_epoch().count();
+      return mtime == dep.getModificationTime();
     }
 
     // Slow path: if the dependency is verified by content hash, check it vs.
     // the hash of the file.
     auto buf = getBufferOfDependency(fs, dep.getPath(), interfacePath,
                                      diags, diagnosticLoc);
-    if (!buf) return None;
+    if (!buf) return false;
 
     if (xxHash64(buf->getBuffer()) == dep.getContentHash())
-      return Status;
+      return true;
 
-    return None;
+    return false;
```

Comment:
Nitpick: up until this point you've been using early returns for the "sad path". I'd suggest switching these two returns to stick with that (or folding it into one line like you did with the mtime).